### DISCUSSION TITLE
How to implement multiple data outputs

### DISCUSSION THREAD

**Discussion by nineswords-z** [JSBSim-Team/jsbsim#1044] [Q&A] [answered]
hi all, I tried to modify the XML file to achieve multi-channel data output, but the XML file parsing failed, how can I modify the XML file? Thank you very much.
xml file:
![Snipaste_2024-02-20_09-38-31](https://github.com/JSBSim-Team/jsbsim/assets/5345756/96851cf8-fe5f-4eda-b2b7-56e843b6315b)

cmd print:
`.\JSBSim.exe --realtime data_output/flightgear2.xml scripts/c1723.xml

In file data_output/flightgear2.xml: line 20
XML parse error: junk after document element
FATAL ERROR: JSBSim terminated with an exception.
The message was: In file data_output/flightgear2.xml: line 20
XML parse error: junk after document element`

**Comment by seanmcleod** [comment]
Just to confirm I can reproduce the issue you're seeing. Although as per @bcoconni's recent work on XML validation he mentions that multiple `<output />` elements are supported at least as part of `fdm_config`.

**Comment by seanmcleod** [comment]
Ah, the XML file you created isn't a valid XML file in general. A single top-level element is required.

https://stackoverflow.com/questions/14388122/multiple-xml-elements-in-the-same-xml-file

**Comment by bcoconni** [comment, marked as answer]
In addition you should be able to supply as many output files to `JSBSim.exe` as you like. Say you have 2 output files `data_output/flightgear.xml` and `data_output/flightgear2.xml` then you can use the following command line:
```pwsh
JSBSim.exe --realtime scripts/c1723.xml data_output/flightgear.xml data_output/flightgear2.xml
```
For your problem you can create 2 files, one containing
```xml
<?xml version="1.0" encoding="utf-8"?>
<output name="localhost" type="FLIGHTGEAR" protocol="UDP" port="5550" rate="60" version="24">
</output>
```
and the other file containing
```xml
<?xml version="1.0" encoding="utf-8"?>
<output name="localhost" type="FLIGHTGEAR" protocol="UDP" port="5550" rate="60" version="24">
</output>
```
and use them as suggested above.

**Reply by nineswords-z** [reply]
Thanks, that's the right answer.
